low-level/files/opt/seagate/sspl/setup/sspl_setup.py

```diff
@@ -539,6 +541,56 @@ def process(self):
         logger.info(f"{self.name} interface not implemented.")
 
 
+class PreUpgradeCmd(Cmd):
+    """PreUpgrade support for SSPL componenet."""
+
+    name = "pre_upgrade"
+
+    def __init__(self, args):
+        super().__init__(args)
+
+    def validate(self):
+        # Common validator classes to check Cortx/system wide validator
+        pass
+
+    def process(self):
+        logger.info(f"{self.name} interface not implemented.")
+
+
+class PostUpgradeCmd(Cmd):
+    """PostUpgrade support for SSPL componenet."""
+
+    name = "post_upgrade"
+
+    def __init__(self, args):
+        super().__init__(args)
+
+    def validate(self):
+        # Common validator classes to check Cortx/system wide validator
+        pass
+
+    def process(self):
+        new_conf_url = 'yaml:///opt/seagate/cortx/sspl/conf/sspl.conf.LR2.yaml'
+        existing_conf_url = 'yaml:///etc/sspl.conf'
+        merged_conf_url = 'yaml:///opt/seagate/cortx/sspl/tmp/merged.conf'
+        # Only proceed if both existing and new config path are present
+        for filepath in [existing_conf_url, new_conf_url]:
+            if not os.path.exists(filepath.split(":/")[1]):
+                logger.info("Exiting, File %s does not exists" % filepath)
+                return
+        conf_upgrade = ConfUpgrade(existing_conf_url, new_conf_url,
+                                   merged_conf_url)
+        try:
+            conf_upgrade.create_merged_config()
+        except ConfError as e:
+            logger.error("Error: %s while upgrading config file, existing config"
```

Comment:
Again message can be better i feel, as these are going to be important if/when upgrade error seen

logger.error(" \"%s\" error seen while upgrading config, existing config retained" % e)

Reply:
Done